docroot/modules/custom/annotation_store/src/AnnotationStoreAccessControlHandler.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Drupal\annotation_store;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Entity\EntityAccessControlHandler;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Session\AccountInterface;
+
+/**
+ * Access controller for the comment entity.
+ *
+ * @see \Drupal\comment\Entity\Comment.
+ */
+class AnnotationStoreAccessControlHandler extends EntityAccessControlHandler {
+
+  /**
+   * {@inheritdoc}
+   *
+   * Link the activities to the permissions. checkAccess is called with the
+   * $operation as defined in the routing.yml file.
+   */
+  protected function checkAccess(EntityInterface $entity, $operation, AccountInterface $account) {
+    switch ($operation) {
+      case 'view':
+        return AccessResult::allowedIfHasPermission($account, 'View annotation store');
+
+    }
+    return AccessResult::allowed();
```

Comment:
Not sure if this is correct. I might be wrong here but, looks like you allowing all the operations except for view without any access check?